### ISSUE TITLE
Catalog cleanup fails with KeyError

### ISSUE THREAD

**Issue by hvelarde** [collective/collective.catalogcleanup#14]
I'm getting the following error while running a catalog cleanup:

```pypdb
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module collective.catalogcleanup.browser, line 68, in __call__
  Module collective.catalogcleanup.browser, line 240, in non_unique_uids
   - __traceback_info__: uid_catalog
  Module collective.catalogcleanup.browser, line 22, in path_len
  Module Products.ZCatalog.CatalogBrains, line 51, in getPath
  Module Products.ZCatalog.ZCatalog, line 518, in getpath
KeyError: 1304083406
```

**Comment by mauritsvanrees** [comment]
We are catching `KeyErrors` for `getPath` in several places already, so I can do that here too. I am working on a fix.

I wonder if we should remove any brains for which this is the case.

Or is this maybe during a dry run? It could be that this happens for brains that would already have been removed earlier in the cleanup if it wasn't a dry run.

**Comment by hvelarde** [comment]
IIRC, this happened with `dry_run=false`.

**Comment by mauritsvanrees** [comment]
Fixed in release 1.8.0.
